src/backends/tensorrt/plan_backend.cc

```diff
@@ -364,10 +453,62 @@ PlanBackend::CreateExecutionContext(
     }
   }
 
-  // Create CUDA stream associated with the execution context
-  const int cuda_stream_priority =
-      GetCudaStreamPriority(Config().optimization().priority());
-  RETURN_IF_ERROR(context->CreateCudaStream(cuda_stream_priority));
+  // Passing the queue for available contexts here so that completion thread
+  // knows where to inform that the context is ready for inputs.
+  std::shared_ptr<SyncQueue<size_t>> lqueue = context_queue;
+  context->completion_thread_ = std::thread([context, context_idx, lqueue]() {
```

Comment:
Completion thread functionality is a little big for a lambda. Can you put it in a static function and use that.